pkg/server/handlers.go

```diff
@@ -11,10 +13,51 @@ import (
 	"github.com/grafana/iot-sitewise-datasource/pkg/models"
 )
 
+func getErrorResponse(err error, message string) *backend.QueryDataResponse {
+	resp := backend.Responses{
+		"A": backend.DataResponse{
+			Error: errors.New(message + ": " + err.Error()),
+		},
+	}
+	return &backend.QueryDataResponse{Responses: resp}
+}
+
 func processQueries(ctx context.Context, req *backend.QueryDataRequest, handler QueryHandlerFunc) *backend.QueryDataResponse {
 	res := backend.Responses{}
-	for _, v := range req.Queries {
-		res[v.RefID] = handler(ctx, req, v)
+
+	var datasourceJsonData struct {
+		DefaultRegion string `json:"defaultRegion"`
+	}
+
+	defaultRegion := ""
+	if req.PluginContext.DataSourceInstanceSettings != nil &&
+		req.PluginContext.DataSourceInstanceSettings.JSONData != nil {
+
+		if err := json.Unmarshal(req.PluginContext.DataSourceInstanceSettings.JSONData, &datasourceJsonData); err != nil {
+			return getErrorResponse(err, "failed to unmarshal datasource JSON data")
+		}
+		if datasourceJsonData.DefaultRegion != "" {
+			defaultRegion = datasourceJsonData.DefaultRegion
```

Comment:
Can this be done here instead https://github.com/grafana/iot-sitewise-datasource/blob/ee4a1f1416fdf92367d4b6550e23512c5d8a8cf4/pkg/sitewise/datasource.go#L115 with something like

```
	if region == "" {
		region = ds.cfg.Region
	}
```

Reply:
i think the `ds.cfg.Region` should already have the defaultRegion set there by the time we reach that code because it's loaded when the settings are loaded https://github.com/grafana/iot-sitewise-datasource/blob/ee4a1f1416fdf92367d4b6550e23512c5d8a8cf4/pkg/models/setting.go#L31-L33

Reply:
Ok I think it should be possible, I thought region was necessary for the query but it isn't. Let me try to rewrite this